src/EnergyPlus/OutputProcessor.cc

```diff
@@ -6468,26 +6466,37 @@ void UpdateMeterReporting(EnergyPlusData &state)
     }
 
     // Helper lambda to locate a meter index from its name. Returns a negative value if not found
-    auto findMeterIndexFromMeterName = [&state](std::string const &name) -> int {
-        auto &op(state.dataOutputProcessor);
+    auto setupMeterFromMeterName =
+        [&state](std::string &name, std::string const &freqString, bool MeterFileOnlyIndicator, bool CumulativeIndicator) -> bool {
+        bool result = false;
 
-        // Return a value <= 0 if not found
-        int meterIndex = -99;
+        auto varnameLen = index(name, '[');
+        if (varnameLen != std::string::npos) {
+            name.erase(varnameLen);
+        }
+
+        auto &op(state.dataOutputProcessor);
 
         std::string::size_type wildCardPosition = index(name, '*');
 
         if (wildCardPosition == std::string::npos) {
-            meterIndex = UtilityRoutines::FindItem(name, op->EnergyMeters);
+            int meterIndex = UtilityRoutines::FindItem(name, op->EnergyMeters);
+            if (meterIndex > 0) {
+                ReportingFrequency ReportFreq = determineFrequency(state, freqString);
+                SetInitialMeterReportingAndOutputNames(state, meterIndex, MeterFileOnlyIndicator, ReportFreq, CumulativeIndicator);
+                result = true;
+            }
         } else { // Wildcard input
-            for (int Meter = 1; Meter <= op->NumEnergyMeters; ++Meter) {
-                if (UtilityRoutines::SameString(op->EnergyMeters(Meter).Name.substr(0, wildCardPosition), name.substr(0, wildCardPosition))) {
-                    meterIndex = Meter;
-                    break;
+            ReportingFrequency ReportFreq = determineFrequency(state, freqString);
+            for (int meterIndex = 1; meterIndex <= op->NumEnergyMeters; ++meterIndex) {
+                if (UtilityRoutines::SameString(op->EnergyMeters(meterIndex).Name.substr(0, wildCardPosition), name.substr(0, wildCardPosition))) {
+                    SetInitialMeterReportingAndOutputNames(state, meterIndex, MeterFileOnlyIndicator, ReportFreq, CumulativeIndicator);
+                    result = true;
                 }
             }
         }
 
-        return meterIndex;
+        return result;
```

Comment:
Refactored the lambda to do more, and so I can call `SetInitialMeterReportingAndOutputNames` in a loop as well